Sources/MetricKit/MetricsMonitor.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MetricsMonitor.swift
+//  
+//
+//  Created by Antoine van der Lee on 31/08/2021.
+//
+
+import Foundation
+import MetricKit
+
+/// Monitors payloads delivered by MetricKit and logs valueable information, including crashes.
+final class MetricsMonitor: NSObject {
+
+    func startMonitoring() {
+        if #available(iOS 14, *) {
+            MXMetricManager.shared.add(self)
+        }
+
+        NSSetUncaughtExceptionHandler { exception in
+            MetricsMonitor.logExceptionUsingCallStackSymbols(exception, description: "Uncaught Exception")
+        }
+    }
+
+    /// Creates a new log section with the current thread call stack symbols.
+    private static func logExceptionUsingCallStackSymbols(_ exception: NSException, description: String) {
+        let message = """
+
+        ---
+
+        🚨 CRASH:
+        Description: \(description)
+        Exception name: \(exception.name.rawValue)
+        Reason: \(exception.reason ?? "nil")
+
+            \(Thread.callStackSymbols.joined(separator: "\n"))
+
+        ---
+
+        """
+        DiagnosticsLogger.standard.log(message)
+    }
+}
+
+extension MetricsMonitor: MXMetricManagerSubscriber {
+    @available(iOS 13.0, *)
+    func didReceive(_ payloads: [MXMetricPayload]) {
+        // We don't do anything with metrics yet.
```

Comment:
Should we open an issue for this? There might be some valuable information we could log as well.

Reply:
I checked whether there was something obvious valuable, but there wasn't. Crashes really is the one thing we need now I think. Battery draining information and stuff won't really help in support issues.

Let me know if you disagree!

Reply:
Yeah I agree. The only one that comes to mind is background termination reason and memory termination.